### PR TITLE
Add superclusters (categories)

### PR DESCRIPTION
**Pull request by dsmilkov**

### CODE REVIEW

File: lilac/data/clustering.py [hunk 1]
```diff
@@ -105,6 +111,37 @@ def summarize_instructions(ranked_docs: list[tuple[str, float]]) -> str:
   return title.title
 
 
+class Category(BaseModel):
+  """A 4-5 word category."""
```
Comment (nsthorat) [inline lilac/data/clustering.py:115]: is this shipped to openai? maybe remove the numbers, "a short category title"

File: lilac/data/clustering.py [hunk 2]
```diff
@@ -113,6 +150,7 @@ def cluster(
   topic_fn: TopicFn = summarize_instructions,
   overwrite: bool = False,
   remote: bool = False,
+  supertopic: bool = False,
```
Comment (nsthorat) [inline lilac/data/clustering.py:153]: maybe category?

File: lilac/data/clustering.py [hunk 3]
```diff
@@ -208,23 +246,93 @@ def _compute_topic(cluster_id: int) -> Optional[str]:
...
 
   # Output the topic as a child of the cluster enrichment.
-  topic_output_path = (*cluster_output_path, TOPIC_FIELD_NAME)
-  dataset.transform(
-    functools.partial(_compute_topics, text_column, cluster_column),
-    input_path=ancestor_path,
-    output_path=topic_output_path,
+  topic_output_path = (*cluster_output_path, CLUSTER_TITLE)
+
+  topics_exist = dataset.manifest().data_schema.has_field(topic_output_path)
```
Comment (nsthorat) [inline lilac/data/clustering.py:251]: maybe use cluster_title instead of topics for consistency

**Review by nsthorat — APPROVED** [review]
amazing work!!

File: lilac/data/clustering.py (hunk 1 above)
Comment (dsmilkov) [reply lilac/data/clustering.py:115]: done

File: lilac/data/clustering.py (hunk 3 above)
Comment (dsmilkov) [reply lilac/data/clustering.py:251]: done, here and elsewhere

File: lilac/data/clustering.py (hunk 2 above)
Comment (dsmilkov) [reply lilac/data/clustering.py]: done

**Pull request merged**